pkg/webhook/preflight/nutanix/imagekubernetesversioncheck.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2025 Nutanix. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package nutanix
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+
+	vmmv4 "github.com/nutanix/ntnx-api-golang-clients/vmm-go-client/v4/models/vmm/v4/content"
+
+	carenv1 "github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/api/v1alpha1"
+	"github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pkg/webhook/preflight"
+)
+
+// The regex captures the Kubernetes version in the format of 1.x.y, where x and y are digits.
+// Example: kubedistro-rocky-9.5-vgpu-1.32.3-20250604180644 -> 1.32.3.
+var kubernetesVersionRegex = regexp.MustCompile(`(?i)\b[vV]?(1\.\d+(?:\.\d+)?)\b`)
```

Comment:
Do we need this regex, would it not be enough to just look for the for `c.clusterK8sVersion` in the `imageName`?
Having minor and patch versions be optional may incorrectly match against a different part of the image string.